### PR TITLE
1.1.1k fips jratest

### PR DESCRIPTION
**Pull request by jallisonciq**
Add in the FIPS-140-3 changes required by the lab.

### CODE REVIEW

File: crypto/rand/rand_lib.c [hunk 1]
```diff
@@ -189,6 +194,7 @@ size_t rand_drbg_get_entropy(RAND_DRBG *drbg,
         }
 
     } else {
+#if 0
```
Comment (jason-rodri) [inline crypto/rand/rand_lib.c:207]: add a comment on why this was commented out please

Comment (jallisonciq) [reply crypto/rand/rand_lib.c:207]: OK.

File: crypto/rand/rand_lib.c [hunk 2]
```diff
@@ -163,6 +163,11 @@ size_t rand_drbg_get_entropy(RAND_DRBG *drbg,
 
         if (buffer != NULL) {
             size_t bytes = 0;
+            RAND_DRBG *master_drbg = RAND_DRBG_get0_master();
+
+            if (drbg->parent == master_drbg) {
+                 prediction_resistance = 1;
```
Comment (jason-rodri) [inline crypto/rand/rand_lib.c:179]: why only for the master?

**Review by jason-rodri — APPROVED** [review]

File: crypto/rand/rand_lib.c (hunk 2 above)
Comment (jallisonciq) [reply crypto/rand/rand_lib.c:179]: Lab request - needed only for master.

"FIPS 140-3 IG D.K Resolution 5. Requires that the master DRBG shall have prediction resistance enabled, or shall be reseeded before requesting an output."

Can't enable prediction resistance as it's not implemented, (hence the #if 0 above) but setting the variable prediction_resistance = 1 when it's the master DRBG forces reseed.

Lab confirmed this code satisfies 140-3.

**Pull request merged**